src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,52 @@
 package core.basesyntax;
 
-class Pair {
+class Pair<F, S> {
+    private F first;
+    private S second;
 
+    private Pair(F first, S second) {
+        this.first = first;
+        this.second = second;
+    }
+
+    public F getFirst() {
+        return first;
+    }
+
+    public S getSecond() {
+        return second;
+    }
+
+    public static <F, S> Pair<F, S> of(F first, S second) {
+        return new Pair<>(first, second);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+
+        if (obj == null) {
+            return false;
+        }
+
+        if (this == obj) {
+            return true;
+        }
+
+        if (obj.getClass().equals(getClass())) {
+            Pair<F, S> pair = (Pair<F, S>) obj;
+
+            return first == null ? pair.first == null : first.equals(pair.first)
```

Comment:
let's make this check without ternary operator, it should be something like this 

```
return (pair.getFirst() == this.first
                || (pair.getFirst() != null && pair.getFirst().equals(this.first)))
                && (pair.getSecond() == this.second
                || (pair.getSecond() != null && pair.getSecond().equals(this.second)));
```